contracts/oracle/PCVOracle.sol

```diff
@@ -0,0 +1,370 @@
+pragma solidity 0.8.13;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {Decimal} from "../external/Decimal.sol";
+
+import {IOracle} from "./IOracle.sol";
+import {VoltRoles} from "../core/VoltRoles.sol";
+import {CoreRefV2} from "../refs/CoreRefV2.sol";
+import {PCVDeposit} from "../pcv/PCVDeposit.sol";
+import {MarketGovernanceOracle} from "./MarketGovernanceOracle.sol";
+
+contract PCVOracle is CoreRefV2 {
+    using Decimal for Decimal.D256;
+    using SafeCast for *;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @notice emitted when a new token oracle is set
+    event OracleUpdate(
+        address indexed token,
+        address indexed oldOracle,
+        address indexed newOracle
+    );
+
+    /// @notice emitted when a new venue is added
+    event VenueAdded(address indexed venue, bool isIliquid, uint256 timestamp);
+
+    /// @notice emitted when a venue is removed
+    event VenueRemoved(
+        address indexed venue,
+        bool isIliquid,
+        uint256 timestamp
+    );
+
+    /// @notice emitted when total venue PCV changes
+    event PCVUpdated(
+        address indexed venue,
+        bool isIliquid,
+        uint256 timestamp,
+        uint256 oldLiquidity,
+        uint256 newLiquidity
+    );
+
+    /// @notice emitted when market governance oracle is updated
+    event MarketGovernanceOracleUpdated(
+        address oldMgovOracle,
+        address newMgovOracle
+    );
+
+    /// @notice Map of oracles to use to get USD values of assets held in
+    ///         PCV deposits. This map is used to get the oracle address from
+    ///         and ERC20 address.
+    mapping(address => address) public tokenToOracle;
+
+    /// @notice Map from deposit address to token address. It is used the oracle
+    /// of a PCVDeposit by using tokenToOracle(depositToToken(depositAddress)).
+    mapping(address => address) public depositToToken;
+
+    ///@notice set of whitelisted pcvDeposit addresses for withdrawal
+    EnumerableSet.AddressSet private liquidVenues;
+
+    ///@notice set of whitelisted pcvDeposit addresses for withdrawal
+    EnumerableSet.AddressSet private illiquidVenues;
+
+    /// @notice reference to the market governance oracle smart contract
+    address public marketGovernanceOracle;
+
+    /// @notice last illiquid balance
+    uint256 public lastIlliquidBalance;
+
+    /// @notice last liquid balance
+    uint256 public lastLiquidBalance;
```

Comment:
Would it make sense to combine these into a single slot, so if we update both liquid and illiquid in a single tx, we only pay for a cold and warm SSTORE  instead of two cold SSTORE's?

Reply:
the call either starts from `updateLiquidBalance` or `updateIlliquidBalance` so I don't think we can save 1 SSTORE here